### DISCUSSION TITLE
general questions about the status of the project

### DISCUSSION THREAD

**Discussion by alex-military** [vortigont/Songguo-PTS200#10] [Q&A]
hello I follow the project for the 120W xiaomi power supply issue on the original firmware project. I wanted to know at what point the firmware development is for the parts that are missing mainly for the calibration of the new tips and for the selection of the power supply power. I also wanted to know if the pwm curve of the power has been tested to prevent power supplies like the xiaomi from entering protection. last thing I wanted if you will do an update method like the original file with a fle bin.

thank you for the work done surely as soon as it is complete I will implement the new firmware.

**Comment by vortigont** [comment]
Hey there!
Project is not abandoned for sure! Just had other private priority things lately.
I'll keep working on this project bit by bit. Tip calibration and more advanced power management is on short todo list.
Currently my top priority is to get rid of dependency on any Arduino core functions and use pure IDF API. I need that to switch to IDF 5.x, for this need to change low level ADC code, etc... check compatibility with other libs.
For power ramping - it's already implemented and works fine for me but I do not have many chargers around to do a comprehensive test. Mostly awaiting some feedback here. It works fine with powerbanks also, at least with 3 that I have in possession.
File updates via drag-and-drop is also planned for sure, but a bit later maybe.

Thanks for your interest here.
